icu4c/source/test/intltest/ucolheaderonlytest.cpp

```diff
@@ -0,0 +1,96 @@
+// © 2025 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// Test header-only ICU C++ APIs. Do not use other ICU C++ APIs.
+// Non-default configuration:
+#define U_SHOW_CPLUSPLUS_API 0
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_COLLATION
+
+#include <string_view>
+
+#include "unicode/ucol.h"
+
+#include "intltest.h"
+
+namespace {
+
+class UColHeaderOnlyTest : public IntlTest {
+public:
+    UColHeaderOnlyTest() = default;
+    void runIndexedTest(int32_t index, UBool exec, const char*& name, char* par = nullptr) override;
+    void TestPredicateTypes();
+};
+
+void UColHeaderOnlyTest::runIndexedTest(int32_t index, UBool exec, const char*& name, char* /*par*/) {
+    if (exec) {
+        logln("TestSuite UColHeaderOnlyTest: ");
+    }
+    TESTCASE_AUTO_BEGIN;
+    TESTCASE_AUTO(TestPredicateTypes);
+    TESTCASE_AUTO_END;
+}
+
+class UCollatorWrapper {
```

Comment:
FYI: We can use std::unique_ptr here.

Reply:
> FYI: We can use std::unique_ptr here.

I can't see how, but please send a follow-up PR if you have some clever idea.

Reply:
You would provide ucol_close as the deleter: https://en.cppreference.com/w/cpp/memory/unique_ptr/unique_ptr
Not important to change.

Reply:
I still can't imagine how that would lead to simpler code, but please send a follow-up PR if you do.